DisCatSharp/Entities/Guild/ScheduledEvent/DiscordScheduledEvent.cs

```diff
@@ -204,35 +204,20 @@ public async Task ModifyAsync(Action<ScheduledEventEditModel> action)
 			var mdl = new ScheduledEventEditModel();
 			action(mdl);
 
-			var channelId = Optional.FromNoValue<ulong?>();
-			if (mdl.Channel.HasValue && (mdl.Channel.Value.Type != ChannelType.Voice || mdl.Channel.Value.Type != ChannelType.Stage) && mdl.Channel.Value != null)
-				throw new ArgumentException("Channel needs to be a voice or stage channel.");
-			else if (mdl.Channel.HasValue && mdl.Channel.Value != null)
-				channelId = mdl.Channel.Value.Id;
+			Optional<ulong?> channelId = null;
+			if (this.EntityType == ScheduledEventEntityType.External || mdl.EntityType != ScheduledEventEntityType.External)
+				channelId = mdl.Channel
+					.MapOrNull<ulong?>(c => c.Type != ChannelType.Voice && c.Type != ChannelType.Stage
+						? throw new ArgumentException("Channel needs to be a voice or stage channel.")
+						: c.Id);
 
-			if (this.EntityType != ScheduledEventEntityType.External && mdl.EntityType == ScheduledEventEntityType.External)
-				channelId = null;
```

Comment:
I would like to raise attention to this change.

Originally, when `mdl.Channel` was `null` this would output `None`, rather than an `Some(null)`. This commit changes this behavior, so that an `Optional` containing `null` maps to an `Optional` containing `null`, as is pretty much the case everywhere else. If this unusual behavior was actually intended, rather than the result of convoluted logic, this would need to be adjusted, preferably with a comment explaining the rationale.

Reply:
I'm totally ok with it!